packages/beacon-node/src/chain/blocks/verifyBlock.ts

```diff
@@ -101,6 +136,7 @@ export async function verifyBlocksInEpoch(
 
       // All signatures at once
       verifyBlocksSignatures(this.bls, this.logger, this.metrics, preState0, blocks, opts),
+      ...persistBlockPromises,
```

Comment:
Move the new code to a function called `eagerlyPersistBlockInput()` and follow similar pattern to the functions above. Do the Promise.all inside that function